lib/rucio/tests/test_api_external_representation.py

```diff
@@ -128,3 +161,257 @@ def test_api_account_limit(self):
         for usage in out:
             if 'rse_expression' in usage:
                 assert_equal(rse_expr, usage['rse_expression'])
+
+    def test_api_did(self):
+        """ DID (API): Test external representation of DIDs """
+        # add some dids
+        add_did(self.scope_name, 'ext_parent', 'container', issuer='root', account=self.account_name, **self.vo)
+        add_did(self.scope_name, 'ext_child', 'dataset', issuer='root', account=self.account_name, **self.vo)
+        attachment = {'scope': self.scope_name, 'name': 'ext_parent',
+                      'dids': [{'scope': self.scope_name, 'name': 'ext_child', 'type': 'DATASET'}]}
+        attach_dids_to_dids([attachment], issuer='root', **self.vo)
+
+        # test scope_list
+        out = scope_list(self.scope_name, recursive=True, **self.vo)
+        for did in out:
+            assert_equal(did['scope'], self.scope_name)
+            if did['parent'] is not None:
+                assert_equal(did['parent']['scope'], self.scope_name)
+
+        # test get_did
+        add_did_to_followed(self.scope_name, 'ext_parent', self.account_name, **self.vo)
+        out = get_users_following_did('ext_parent', self.scope_name, **self.vo)
+        for user in out:
+            assert_equal(user['user'], self.account_name)
+
+    def test_api_exporter(self):
+        """ EXPORTER (API): Test external representation of exported data """
+
+        out = export_data('root', **self.new_vo)
+        rses = out['rses']
+        assert_in(self.rse3_name, rses)
+        assert_not_in(self.rse3_id, rses)
+
+        distances = out['distances']
+        assert_in(self.rse3_name, distances)
+        assert_not_in(self.rse3_id, distances)
+        assert_in(self.rse4_name, distances[self.rse3_name])
+        assert_not_in(self.rse4_id, distances[self.rse3_name])
+
+        # check for interference from other VOs
+        if self.multi_vo:
+            assert_not_in(self.rse_name, rses)
+            assert_not_in(self.rse_id, rses)
+            assert_not_in(self.rse2_name, rses)
+            assert_not_in(self.rse2_id, rses)
+
+            assert_not_in(self.rse_name, distances)
+            assert_not_in(self.rse_id, distances)
+            assert_not_in(self.rse2_name, distances)
+            assert_not_in(self.rse2_id, distances)
+
+    def test_api_identity(self):
+        """ IDENTITY (API): Test external representation of identity accounts """
+
+        id_key = ''.join(random.choice(string.ascii_lowercase) for x in range(10))
+
+        add_account_identity(id_key, 'userpass', self.account_name, 'rucio_test@test.com', 'root', default=True, password='ext_pass', **self.vo)
+
+        # TODO - fix passing default to add_account_identity
```

Comment:
Does this TODO/fix need its own issue creating or will it be covered by the existing issue (#25).

Reply:
This was something I discussed with Martin last week, whether the 'default' argument is even needed in the code any more. Since the conclusion was that it's not really needed, and it's not accessible from the higher layers, I'll take this test out.